### PR TITLE
Add bitcoin address form field

### PR DESCRIPTION
**Pull request by ivan-zp-ess**

### CODE REVIEW

File: crypto_django/forms/fields/addresses/bitcoin.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+"""
+Provide implementation for Ethereum address form field.
```
Comment (dmytrostriletskyi) [inline crypto_django/forms/fields/addresses/bitcoin.py]: `Ethereum` > `Bitcoin`.

File: crypto_django/forms/fields/addresses/bitcoin.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+)
+from crypto_django.forms.fields.bech32 import Bech32
+
+
+class BitcoinAddressField(forms.CharField):
+    """
+    Bitcoin address form field implementation.
+    """
+
+    default_error_messages = {  # BITCOIN
```
Comment (dmytrostriletskyi) [inline crypto_django/forms/fields/addresses/bitcoin.py]: Remove `# BITCOIN`.

File: crypto_django/forms/fields/addresses/bitcoin.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        'prefix': 'Invalid address prefix - it has to start with one of the [1, 3, bc1]',
+    }
+
+    def to_python(self, value):
+        address = value
+
+        if address.startswith(BECH32_PREFIX):
+            Bech32().bech32_decode_check(address)
+        elif address.startswith(P2SH_PREFIX) or address.startswith(P2PKH_PREFIX):
+            b58decode_check(value)
```
Comment (dmytrostriletskyi) [inline crypto_django/forms/fields/addresses/bitcoin.py]: You should make validation flow as in [Ethereum field](https://github.com/essentiaone/crypto-django/blob/6ae41b567f09ea36e0d9e97fd08a92feb369fdff/crypto_django/forms/fields/addresses/ethereum.py#L50). Also without `elif`, please.

```python
if not address.startswith(BECH32_PREFIX):
    raise

if not Bech32().bech32_decode_check(address):
    raise

...
```

Is it possible?

File: crypto_django/forms/fields/bech32.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
+class Bech32:
```
Comment (dmytrostriletskyi) [inline crypto_django/forms/fields/bech32.py]: Create module docstring, please.

File: crypto_django/forms/fields/bech32.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    """
+    Algorithm implementation for checking segwit address validity
+
+    References:
+        - github.com/bitcoin/bips/blob/master/bip-0173.mediawiki
+        - github.com/sipa/bech32/blob/master/ref/python/segwit_addr.py
+    """
+    CHARSET = "qpzry9x8gf2tvdw0s3jn54khce6mua7l"
```
Comment (dmytrostriletskyi) [inline crypto_django/forms/fields/bech32.py]: Outside of class, please.

File: crypto_django/forms/fields/bech32.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+    def bech32_hrp_expand(self, hrp):
+        """Expand the HRP into values for checksum computation."""
```
Comment (dmytrostriletskyi) [inline crypto_django/forms/fields/bech32.py]: Follow out docstring style, please.

File: requirements.txt [hunk 7]
```diff
@@ -1,3 +1,4 @@
 Django==2.0.7
 eth-utils==1.0.3
 eth-hash[pycryptodome]
+bit
```
Comment (dmytrostriletskyi) [inline requirements.txt]: Add version, please.

File: crypto_django/forms/fields/bech32.py [hunk 8]
```diff
@@ -0,0 +1,50 @@
+class Bech32:
+    """
+    Algorithm implementation for checking segwit address validity
```
Comment (dmytrostriletskyi) [inline crypto_django/forms/fields/bech32.py]: Dot to the end of the sentence.

File: crypto_django/forms/fields/addresses/bitcoin.py (hunk 3 above)
Comment (ivan-zp-ess) [reply crypto_django/forms/fields/addresses/bitcoin.py]: Sure, but this seems to be as good a place to use elif as any :c

File: crypto_django/forms/fields/addresses/bitcoin.py [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+            raise ValidationError(self.error_messages.get('bech32'), code='bech32')
+
+        if address.startswith(P2SH_PREFIX) or address.startswith(P2PKH_PREFIX):
+            try:
+                b58decode_check(value)
+            except ValueError as error_message:
+                error_message_params = {
+                    'b58decode_error': error_message,
+                }
+                raise ValidationError(self.error_messages.get('p2'), code='p2', params=error_message_params)
```
Comment (dmytrostriletskyi) [inline crypto_django/forms/fields/addresses/bitcoin.py]:
```python
if not or not .. :
    raise

try:
    ...

except: 
    raise
```

File: crypto_django/forms/fields/addresses/bitcoin.py [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+
+        References:
+            - github.com/bitcoin/bips/blob/master/bip-0173.mediawiki
+            - github.com/ofek/bit/blob/e5640cbc79c183b8c4051b46b3ccf3e813e021d3/bit/base58.py
+            - github.com/sipa/bech32/blob/master/ref/python/segwit_addr.py
+        """
+        address = value
+        address_length = len(address)
+
+        if address_length < 26:
```
Comment (dmytrostriletskyi) [inline crypto_django/forms/fields/addresses/bitcoin.py]: `26` to constant as in Ethereum.

File: crypto_django/forms/fields/bech32.py (hunk 4 above)
Comment (ivan-zp-ess) [reply crypto_django/forms/fields/bech32.py]: Done

File: crypto_django/forms/fields/addresses/bitcoin.py (hunk 9 above)
Comment (ivan-zp-ess) [reply crypto_django/forms/fields/addresses/bitcoin.py]: Kinda done

File: crypto_django/forms/fields/bech32.py [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+        """
+        return self.bech32_polymod(self.bech32_hrp_expand(hrp) + data) == 1
+
+    def bech32_decode(self, bech):
+        """
+        Validate a Bech32 string, and determine HRP and data.
+        """
+        if ((any(ord(x) < 33 or ord(x) > 126 for x in bech)) or
+                (bech.lower() != bech and bech.upper() != bech)):
+            return (None, None)
```
Comment (anastasiia-bilova) [inline crypto_django/forms/fields/bech32.py]: Divide `if` statements with blank line for visual indent.

**Pull request merged**